pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1229,56 +1238,77 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 		metrics.DurationAppliedToGroupSyncing.Observe(float64(d.Milliseconds()))
 		klog.V(2).Infof("Finished syncing AppliedToGroup %s. (%v)", key, d)
 	}()
-	var pods []*v1.Pod
 	appGroupNodeNames := sets.String{}
 	appliedToGroupObj, found, _ := n.appliedToGroupStore.Get(key)
 	if !found {
 		klog.V(2).Infof("AppliedToGroup %s not found.", key)
 		return nil
 	}
-	memberSetByNode := make(map[string]controlplane.GroupMemberSet)
-	scheduledPodNum, scheduledExtEntityNum := 0, 0
 	appliedToGroup := appliedToGroupObj.(*antreatypes.AppliedToGroup)
-	pods, externalEntities := n.getAppliedToWorkloads(appliedToGroup)
-	for _, pod := range pods {
-		if pod.Spec.NodeName == "" || pod.Spec.HostNetwork == true {
-			// No need to process Pod when it's not scheduled.
-			// HostNetwork Pods will not be applied to by policies.
-			continue
+	memberSetByNode := make(map[string]controlplane.GroupMemberSet)
+	var updatedAppliedToGroup *antreatypes.AppliedToGroup
+	if appliedToGroup.Service != nil {
+		// AppliedToGroup for NodePort Service span to all Nodes.
+		nodeList, err := n.nodeLister.List(labels.Everything())
```

Comment:
It seems if a new Node joins the cluster, its agent wouldn't receive this AppliedToGroup and corresponding policies.

Reply:
Good catch. Added logic handling the Node Add/Delete event.